ntp-daemon/src/peer.rs

```diff
@@ -59,13 +65,16 @@ pub async fn start_peer<A: ToSocketAddrs, C: 'static + NtpClock + Send>(
                         .as_mut()
                         .reset(Instant::now() + poll_interval.as_system_duration());
 
-                    // TODO: Figure out proper error behaviour here
-                    let ntp_instant = NtpInstant::from_ntp_timestamp(clock.now().unwrap());
+                    let ntp_instant = NtpInstant::now();
                     let packet = peer.generate_poll_message(ntp_instant);
+
+                    // TODO error handling here; what if we cannot determine the origin timestamp?
+                    last_send_timestamp = clock.now().ok();
```

Comment:
If we can't properly stamp a send timestamp, that pretty much inhibits any synchronization, so in my view crashing out here with an expect is acceptable. The only other alternative I see is to log the error and not send the packet, but given the type of causes for the clock failing I don't really see this doing much in terms of resilience (it is pretty much guaranteed to always fail if it fails once)